pkg/certificate/manager.go

```diff
@@ -129,34 +137,113 @@ func (m *Manager) handleMRCEvent(mrcClient MRCClient, event MRCEvent) error {
 		}
 
 		c := &issuer{Issuer: client, ID: mrc.Name, CertificateAuthority: ca, TrustDomain: mrc.Spec.TrustDomain}
-		switch {
-		case mrc.Status.State == constants.MRCStateActive:
+		switch mrc.Status.State {
+		case constants.MRCStateActive:
 			m.mu.Lock()
 			m.signingIssuer = c
 			m.validatingIssuer = c
 			m.mu.Unlock()
-		case mrc.Status.State == constants.MRCStateIssuingRollback || mrc.Status.State == constants.MRCStateIssuingRollout:
+		case constants.MRCStateIssuingRollout:
 			m.mu.Lock()
 			m.signingIssuer = c
 			m.mu.Unlock()
-		case mrc.Status.State == constants.MRCStateValidatingRollback || mrc.Status.State == constants.MRCStateValidatingRollout:
+		case constants.MRCStateIssuingRollback, constants.MRCStateValidatingRollout:
 			m.mu.Lock()
 			m.validatingIssuer = c
 			m.mu.Unlock()
+		case constants.MRCStateValidatingRollback:
+			// do nothing, the issuer of active MRC should be used
 		default:
 			m.mu.Lock()
 			m.signingIssuer = c
 			m.validatingIssuer = c
 			m.mu.Unlock()
 		}
 	case MRCEventUpdated:
-		// TODO
+		newMRC := event.NewMRC
+		oldMRC := event.OldMRC
+		if newMRC.Status.State != oldMRC.Status.State {
+			// TODO(jaellio): Do I need to validate status updates here to determine if they are allowed?
```

Comment:
are there state transitions that are not allowed?  How does it get to that?

Reply:
The MRC status must be updated in the following order (note that at any time the state can be set to error): "", validatingRollout, issuingRollout, active, issuingRollback, validatingRollback, inactive. The validating webhook should enforce this behavior since the status at this point can only be updated manually and not by OSM. We should never have to handle MRC event here for invalid status changes but it is possible the validator has failed or a MRC has been updated before the validating server has started. The TODO here is to question whether or not we should perform redundant validation

Reply:
Even though validation *should* catch it in the webhook, I would lead towards doing final validation to ensure we are not moving into a bad workflow state  (i.e moving from `validatingrollout` directly to `active` with out first doing `issuingrollout` if that is the statemachine we are enforcing.

Reply:
Okay, I agree. I'll make those changes one the MRC validation PR is merged.

Reply:
this https://github.com/openservicemesh/osm/pull/4811?